### PR TITLE
refactor: changed QRCodeModal to be normal component instead of modal

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

After a few tries, I found out that the modal created in PR #1339 is not the most efficient way to have the QR code scan. That is why we modified the component to be a normal component to be called as a step in the Send Wizard Modal.

# Changes

- `QRCodeModal` was renamed `QRCodeScan`.
- The code of the component is now the replica of a normal step of a generic Wizard Modal.

# Tests

Created small test to check if the timeout was correctly triggered.

# Note

We decided to dispatch events to navigate back, in accordance to the other steps of the Wizard Modal.

### CODE REVIEW

File: src/frontend/src/lib/components/send/QRCodeScan.svelte [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+
+	export let expectedToken: Token;
+	export let destination: string | undefined;
+	export let amount: number | undefined;
+
+	const dispatch = createEventDispatcher();
+
+	let timeoutId: NodeJS.Timeout | undefined;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/QRCodeScan.svelte:18]: Given that this PR aims to be a refactoring only, can you provide the feature `timeout`, which seems to be a newest thing,  and its motivation in a separate PR please?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/QRCodeScan.svelte]: done

File: src/frontend/src/lib/i18n/en.json [hunk 2]
```diff
@@ -244,7 +244,8 @@
 			"data_undefined": "Transaction Data cannot be undefined or null.",
 			"no_identity_calculate_fee": "No identity provided to calculate the fee for its principal.",
 			"invalid_destination": "The destination is invalid. Please try again with a valid wallet address or destination.",
-			"incompatible_token": "The token is incompatible. Please try again with a compatible token."
+			"incompatible_token": "The token is incompatible. Please try again with a compatible token.",
+			"qr_scan_timeout": "No QR code found. Please try again."
```
Comment (peterpeterparker) [inline src/frontend/src/lib/i18n/en.json:248]: This label should as well be provided in another PR related to the timeout

File: src/frontend/src/tests/lib/components/send/QRCodeScan.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const expectedToken = ICP_TOKEN;
+
+vi.mock('$lib/stores/toasts.store');
+
+describe('QRCodeReaderComponent', () => {
+	afterEach(() => {
+		vi.clearAllMocks();
+	});
+
+	it('should show timeout error on QR scan timeout', async () => {
```
Comment (peterpeterparker) [inline src/frontend/src/tests/lib/components/send/QRCodeScan.spec.ts:17]: ditto
